src/Swipeable.js

```diff
@@ -93,22 +75,34 @@ export default class Swipeable extends PureComponent {
     }
   }
 
-  render() {
-    const { children } = this.props;
-
-    return children(el => {
-      this.el = el;
-    });
+  instanceRef.current = {
+    ...instanceRef.current,
+    handleTouchStart,
+    handleTouchMove,
+    handleTouchEnd,
+  };
+  if (ref) {
+    ref.current = instanceRef.current;
   }
+
+  useEffect(() => {
+    let { el } = instanceRef.current;
+    el.addEventListener("touchstart", handleTouchStart);
+    el.addEventListener("touchmove", handleTouchMove);
+    el.addEventListener("touchend", handleTouchEnd);
+
+    return () => {
+      el.removeEventListener("touchstart", handleTouchStart);
+      el.removeEventListener("touchmove", handleTouchMove);
+      el.removeEventListener("touchend", handleTouchEnd);
+      instanceRef.current = null;
+      if (ref) ref.current = null;
+    };
+  }, []);
```

Comment:
Isn't React going to complain about `ref` and `handleTouchEnd` missing in the dependency array?

Reply:
I fixed the dependency warning. I did not use the `eslint` before.